#Import Pandas, Numpy, Set path, Assign DF


In [1]:
#Import libraries
import pandas as pd
import numpy as np
import os

In [2]:
path = r'/Users/ECC/Career Foundry/Anaconda Python CF Exercises'

In [3]:
path

'/Users/ECC/Career Foundry/Anaconda Python CF Exercises'

In [6]:
df = pd.read_csv(r'/Users/ECC/Career Foundry/Anaconda Python CF Exercises/02 Data/Original Data/orders.csv', index_col = False)

In [7]:
df

order_id  user_id eval_set  order_number  order_dow  \
0         2539329        1    prior             1          2   
1         2398795        1    prior             2          3   
2          473747        1    prior             3          3   
3         2254736        1    prior             4          4   
4          431534        1    prior             5          4   
...           ...      ...      ...           ...        ...   
3421078   2266710   206209    prior            10          5   
3421079   1854736   206209    prior            11          4   
3421080    626363   206209    prior            12          1   
3421081   2977660   206209    prior            13          1   
3421082    272231   206209    train            14          6   

         order_hour_of_day  days_since_prior_order  
0                        8                     NaN  
1                        7                    15.0  
2                       12                    21.0  
3                        7                    29.0  
4                       15                    28.0  
...                    ...                     ...  
3421078                 18                    29.0  
3421079                 10                    30.0  
3421080                 12                    18.0  
3421081                 12                     7.0  
3421082                 14                    30.0  

[3421083 rows x 7 columns]

In [8]:
df.columns

Index(['order_id', 'user_id', 'eval_set', 'order_number', 'order_dow',
       'order_hour_of_day', 'days_since_prior_order'],
      dtype='object')

In [10]:
vars_list = ['order_id', 'user_id', 'order_number', 'order_dow',
       'order_hour_of_day', 'days_since_prior_order']

In [11]:
vars_list

['order_id',
 'user_id',
 'order_number',
 'order_dow',
 'order_hour_of_day',
 'days_since_prior_order']

#Setting vars_list and checking is eval is excluded.


In [12]:
df.head()

order_id  user_id eval_set  order_number  order_dow  order_hour_of_day  \
0   2539329        1    prior             1          2                  8   
1   2398795        1    prior             2          3                  7   
2    473747        1    prior             3          3                 12   
3   2254736        1    prior             4          4                  7   
4    431534        1    prior             5          4                 15   

   days_since_prior_order  
0                     NaN  
1                    15.0  
2                    21.0  
3                    29.0  
4                    28.0

#First 5 rows called using df.head()

In [13]:
df_prods = pd.read_csv(r'/Users/ECC/Career Foundry/Anaconda Python CF Exercises/02 Data/Original Data/products.csv', index_col = False)

In [14]:
df_prods

product_id                                       product_name  \
0               1                         Chocolate Sandwich Cookies   
1               2                                   All-Seasons Salt   
2               3               Robust Golden Unsweetened Oolong Tea   
3               4  Smart Ones Classic Favorites Mini Rigatoni Wit...   
4               5                          Green Chile Anytime Sauce   
...           ...                                                ...   
49688       49684          Vodka, Triple Distilled, Twist of Vanilla   
49689       49685                 En Croute Roast Hazelnut Cranberry   
49690       49686                                   Artisan Baguette   
49691       49687         Smartblend Healthy Metabolism Dry Cat Food   
49692       49688                             Fresh Foaming Cleanser   

       aisle_id  department_id  prices  
0            61             19     5.8  
1           104             13     9.3  
2            94              7     4.5  
3            38              1    10.5  
4             5             13     4.3  
...         ...            ...     ...  
49688       124              5     5.3  
49689        42              1     3.1  
49690       112              3     7.8  
49691        41              8     4.7  
49692        73             11    13.5  

[49693 rows x 5 columns]

#Imported products with code from 6, changed df_prods and products.csv


In [19]:
df_prods.head(20), df_prods.tail(35)

(    product_id                                       product_name  aisle_id  \
 0            1                         Chocolate Sandwich Cookies        61   
 1            2                                   All-Seasons Salt       104   
 2            3               Robust Golden Unsweetened Oolong Tea        94   
 3            4  Smart Ones Classic Favorites Mini Rigatoni Wit...        38   
 4            5                          Green Chile Anytime Sauce         5   
 5            6                                       Dry Nose Oil        11   
 6            7                     Pure Coconut Water With Orange        98   
 7            8                  Cut Russet Potatoes Steam N' Mash       116   
 8            9                  Light Strawberry Blueberry Yogurt       120   
 9           10     Sparkling Orange Juice & Prickly Pear Beverage       115   
 10          11                                  Peach Mango Juice        31   
 11          12                         

#Combined df.head and df.tail functions to call top 20 and bottom 35

In [20]:
df_prods.dtypes, df_prods.shape

(product_id         int64
 product_name      object
 aisle_id           int64
 department_id      int64
 prices           float64
 dtype: object,
 (49693, 5))

#Combined dtypes with shape, but this is not so readable.

In [21]:
df_prods.shape

(49693, 5)

#Looking back at my examples above, I continued to use df for the analysis, instead of df_prods. Went back,  corrected those lines.

In [22]:
df_prods.describe()

product_id      aisle_id  department_id        prices
count  49693.000000  49693.000000   49693.000000  49693.000000
mean   24844.345139     67.770249      11.728433      9.994136
std    14343.717401     38.316774       5.850282    453.519686
min        1.000000      1.000000       1.000000      1.000000
25%    12423.000000     35.000000       7.000000      4.100000
50%    24845.000000     69.000000      13.000000      7.100000
75%    37265.000000    100.000000      17.000000     11.200000
max    49688.000000    134.000000      21.000000  99999.000000

#The max value for Aisle is 134 (found with .describe), department_id is INT64 (as seen above in d.types)